pkg/utils/metrics.go

```diff
@@ -47,31 +47,27 @@ func init() {
 	prometheus.MustRegister(executionDurationGauge)
 }
 
-func RecordMetrics(instanceSuccesses map[string]int, instanceDurations map[string]time.Duration) {
+func RecordMetrics(instance string, status int, duration time.Duration) {
 	const INTEGRATION = "vault-manager"
 
-	for instance, success := range instanceSuccesses {
-		lastReconcileSuccessGauge.With(
-			prometheus.Labels{
-				"address":     instance,
-				"integration": INTEGRATION,
-			}).Set(float64(success))
+	lastReconcileSuccessGauge.With(
+		prometheus.Labels{
+			"address":     instance,
```

Comment:
i just noticed that you name the labels that hold the instance address `address` instead of `shard_id`. our default alerts group by `shard_id`.